sources/packages/backend/apps/api/src/route-controllers/program-year/_tests_/e2e/program-year.students.controller.getProgramYears.e2e-spec.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { ProgramYear } from "@sims/sims-db";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getStudentToken,
+  FakeStudentUsersTypes,
+} from "../../../../testHelpers";
+
+describe("ProgramYearStudentsController(e2e)-getProgramYears", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let programYear2000: ProgramYear;
+  let programYear2024: ProgramYear;
+  let programYear2025: ProgramYear;
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    appDataSource = dataSource;
+    const programYearRepo = appDataSource.getRepository(ProgramYear);
+
+    programYear2000 = {
+      programYear: "2000-2001",
+      programYearDesc:
+        "Study starting between August 01, 2000 and July 31, 2001",
+      formName: "SFAA2000-01",
+      active: false,
+      parentFormName: "SFAA2000-01-parent",
+      partnerFormName: "SFAA2000-01-partner",
+      programYearPrefix: "2000",
+    } as ProgramYear;
+
+    programYear2024 = {
+      programYear: "2024-2025",
+      programYearDesc:
+        "Study starting between August 01, 2024 and July 31, 2025",
+      formName: "SFAA2024-25",
+      active: true,
+      parentFormName: "SFAA2024-01-parent",
+      partnerFormName: "SFAA2024-01-partner",
+      programYearPrefix: "2024",
+    } as ProgramYear;
+
+    programYear2025 = {
+      programYear: "2025-2026",
+      programYearDesc:
+        "Study starting between August 01, 2025 and July 31, 2026",
+      formName: "SFAA2025-26",
+      active: true,
+      parentFormName: "SFAA2025-01-parent",
+      partnerFormName: "SFAA2025-01-partner",
+      programYearPrefix: "2025",
+    } as ProgramYear;
+
+    const programYear2000Promise = programYearRepo.save(programYear2000);
+    const programYear2024Promise = programYearRepo.save(programYear2024);
+    const programYear2025Promise = programYearRepo.save(programYear2025);
+
+    [programYear2000, programYear2024, programYear2025] = await Promise.all([
+      programYear2000Promise,
+      programYear2024Promise,
+      programYear2025Promise,
+    ]);
+
+    app = nestApplication;
+  });
+
+  it("Should get the list of the available program years when available.", async () => {
+    // Arrange
+    const endpoint = "/students/program-year/options-list";
```

Comment:
We have some tests (in general, not talking about this PR) using the `endpoint` and `token` directly at the `Act/Assert`.
I believe that it looks better here at the `Arrange` level as it is right here.
Can we agree to always have the target `endpoint` and the `token` declared at this level?
@ann-aot @guru-aot @dheepak-aot @sh16011993?